### PR TITLE
Fix method camera_to_sky for real events

### PR DESCRIPTION
**Pull request by JouvinLea**
This PR fix the problem in the method computing camera_to_sky for event to take properly into account the time of the observation in the computation.

For the moment, just use time if provided and the default obs time in utils if not printing a warning.

I think the solution to use HA/DEC for MC event instead of RA/DEC should be done in another method when you compute ra/dec for those SkyCoords. For example, if I read correctly to compute the HA from RA, we need to extract the sideral time from the obs time and the location.
```obstime = sky_coords.obstime.value
scale_time = sky_coords.obstime.scale
longitude = sky_coords.location.longitude
latitude = sky_coords.location.latitude
time=Time(obstime, scale, location=(longitude, latitude)
HA = sky_coords.icrs.ra.to("hourangle") -time.sidereal_time("apparent")
dec = sky_coords.icrs.dec
```

Also I need to improve the test to add a check on a ra/dec value when we add the time.

### CODE REVIEW

File: lstchain/reco/utils.py [hunk 1]
```diff
@@ -203,16 +203,24 @@ def reco_source_position_sky(cog_x, cog_y, disp_dx, disp_dy, focal_length, point
     return camera_to_sky(src_x, src_y, focal_length, pointing_alt, pointing_az)
 
 
-def camera_to_sky(pos_x, pos_y, focal, pointing_alt, pointing_az):
+def camera_to_sky(pos_x, pos_y, focal, pointing_alt, pointing_az, time = None):
```
Comment (rlopezcoto) [inline lstchain/reco/utils.py:206]: maybe we should change the name of the function. As @MaxNoe pointed out in #420 , this method is converting to a `HorizonFrame`

File: lstchain/reco/utils.py [hunk 2]
```diff
@@ -200,12 +200,12 @@ def reco_source_position_sky(cog_x, cog_y, disp_dx, disp_dy, focal_length, point
     sky frame: `astropy.coordinates.sky_coordinate.SkyCoord`
     """
     src_x, src_y = disp.disp_to_pos(disp_dx, disp_dy, cog_x, cog_y)
-    return camera_to_sky(src_x, src_y, focal_length, pointing_alt, pointing_az)
+    return camera_to_altaz_frame(src_x, src_y, focal_length, pointing_alt, pointing_az)
 
 
-def camera_to_sky(pos_x, pos_y, focal, pointing_alt, pointing_az, time = None):
+def camera_to_altaz_frame(pos_x, pos_y, focal, pointing_alt, pointing_az, time = None):
```
Comment (MaxNoe) [inline lstchain/reco/utils.py:206]: The frame in the end is a little strange (Either its `camera_frame_to_altaz_frame` or `camera_to_altaz`).
I would just leave out the frame.

`time` should be `obstime` to be consistent with the astropy API, and then you also don't need the `if` statement below.

File: lstchain/reco/utils.py [hunk 3]
```diff
@@ -227,9 +235,16 @@ def camera_to_sky(pos_x, pos_y, focal, pointing_alt, pointing_az):
     focal = 28*u.m
     pointing_alt = np.array([1.0, 1.0]) * u.rad
     pointing_az = np.array([0.2, 0.5]) * u.rad
-    sky_coords = utils.camera_to_sky(pos_x, pos_y, focal, pointing_alt, pointing_az)
+    sky_coords = utils.camera_to_altaz_frame(pos_x, pos_y, focal, pointing_alt, pointing_az)
 
     """
+
+    if time:
```
Comment (MaxNoe) [inline lstchain/reco/utils.py:242]: This if is not needed. You use `obstime=obstime` and it work fine also in the case where `obstime is None`

File: lstchain/reco/utils.py [hunk 4]
```diff
@@ -200,24 +200,32 @@ def reco_source_position_sky(cog_x, cog_y, disp_dx, disp_dy, focal_length, point
...
+    pointing_az: `~astropy.units.Quantity`
+        pointing altitude in angle unit
+    time: `~astropy.time.Time`
+
 
     Returns
     -------
-    sky frame: `astropy.coordinates.sky_coordinate.SkyCoord`
-
+    sky frame: `astropy.coordinates.AltAz`
```
Comment (MaxNoe) [inline lstchain/reco/utils.py:227]: Actually the return value is a `astropy.coordinates.SkyCoordinate` that has a frame of `AltAz`.

So maybe write ``` `astropy.coordinates.SkyCoord` in AltAz frame` ```

File: lstchain/reco/utils.py [hunk 5]
```diff
@@ -200,24 +200,32 @@ def reco_source_position_sky(cog_x, cog_y, disp_dx, disp_dy, focal_length, point
...
+        X coordinate in camera (distance)
+    pos_y: `~astropy.units.Quantity`
+        Y coordinate in camera (distance)
+    focal: `~astropy.units.Quantity`
+        telescope focal (distance)
+    pointing_alt: `~astropy.units.Quantity`
+        pointing altitude in angle unit
+    pointing_az: `~astropy.units.Quantity`
+        pointing altitude in angle unit
+    time: `~astropy.time.Time`
```
Comment (MaxNoe) [inline lstchain/reco/utils.py:222]: Call it obstime

File: lstchain/reco/utils.py (hunk 2 above)
Comment (JouvinLea) [reply lstchain/reco/utils.py]: Thanks @MaxNoe for your usefull input.
Here I disagree with you but may I misunderstood. I think for MC events we need to use the obstime_mc in order to be able to create a posteriori a ra/dec for MC event in HA/dec as proposed @moralejo . To perform that, t I understood we wiil need:
```scale_time = sky_coords.obstime.scale
longitude = sky_coords.location.longitude
latitude = sky_coords.location.latitude
time=Time(obstime, scale, location=(longitude, latitude)
HA = sky_coords.icrs.ra.to("hourangle") -time.sidereal_time("apparent")
dec = sky_coords.icrs.dec
```

So we need to have a false obstime to be able to then transform to HA.

File: lstchain/reco/utils.py [hunk 6]
```diff
@@ -43,8 +44,8 @@
 
 # position of the LST1
 location = EarthLocation.from_geodetic(-17.89139 * u.deg, 28.76139 * u.deg, 2184 * u.m)
-obstime = Time('2018-11-01T02:00')
-horizon_frame = AltAz(location=location, obstime=obstime)
+obstime_mc = Time('2018-11-01T02:00')
```
Comment (MaxNoe) [inline lstchain/reco/utils.py:47]: Just remove this.

Why was this added anyway? What is so special about 2018-11-01T02:00 that it shoud be used as time fo mc?

File: lstchain/reco/utils.py [hunk 7]
```diff
@@ -227,9 +236,16 @@ def camera_to_sky(pos_x, pos_y, focal, pointing_alt, pointing_az):
...
     pointing_az = np.array([0.2, 0.5]) * u.rad
-    sky_coords = utils.camera_to_sky(pos_x, pos_y, focal, pointing_alt, pointing_az)
+    sky_coords = utils.camera_to_altaz_frame(pos_x, pos_y, focal, pointing_alt, pointing_az)
 
     """
+
+    if obstime:
+        horizon_frame = AltAz(location=location, obstime=obstime)
+    else:
+        horizon_frame = AltAz(location=location, obstime=obstime_mc)
```
Comment (MaxNoe) [inline lstchain/reco/utils.py:246]: Again: just don't use an obstime for mc please. It has no meaning.

File: lstchain/reco/utils.py (hunk 2 above)
Comment (MaxNoe) [reply lstchain/reco/utils.py]: I don't think @moralejo was proposing to actually calculate HA, just mentioning the possibility.

At the moment, there is no usecase I know of for having simulations in another frame than horizon.

A fake time however would never be the solution!

File: lstchain/reco/utils.py [hunk 8]
```diff
@@ -239,12 +239,9 @@ def camera_to_altaz(pos_x, pos_y, focal, pointing_alt, pointing_az, obstime = No
     sky_coords = utils.camera_to_altaz_frame(pos_x, pos_y, focal, pointing_alt, pointing_az)
 
     """
-
-    if obstime:
-        horizon_frame = AltAz(location=location, obstime=obstime)
-    else:
-        horizon_frame = AltAz(location=location, obstime=obstime_mc)
+    if not obstime:
         logging.warning("No time given, use default observation time. To be use only for MC data.")
```
Comment (MaxNoe) [inline lstchain/reco/utils.py]: Now, you don't use any obstime (obstime is None, you are not using the global)

File: lstchain/reco/utils.py (hunk 2 above)
Comment (MaxNoe) [reply lstchain/reco/utils.py]: I think that the monte carlo processing should be fully agnostic of any obstime. Introducing a fake obstime is a hack (that should be unnecessary).

If it at some position is necessary to introduce a fake time, I would be glad to know, because that means we nead it fixed.

**Comment by MaxNoe** [comment]
Btw. the frame to get local hourangle and declination would be the Celestial Intermediate Reference System:
https://docs.astropy.org/en/stable/api/astropy.coordinates.CIRS.html

File: lstchain/reco/utils.py (hunk 8 above)
Comment (JouvinLea) [reply lstchain/reco/utils.py]: yes this is what you ask no?

Comment (MaxNoe) [reply lstchain/reco/utils.py]: But the logging message says something about a default obstime, you not using any obstime

Comment (MaxNoe) [reply lstchain/reco/utils.py]: And I would say, as this is the expected situation for every simulated event, this should not be a "warning".

Warnings should be for unexpected conditions that require the user to do something. Meaningless warnings are dangerous because then users start ignoring them and potentially start ignoring also the important warnings.

Comment (JouvinLea) [reply lstchain/reco/utils.py]: This is because you are too fast, I already modify the print in the time you did your comment

Comment (JouvinLea) [reply lstchain/reco/utils.py]: I agree for the warning

**Comment by MaxNoe** [comment]
There is an open issue for requesting the local equatorial frame in astropy here:
https://github.com/astropy/astropy/issues/10239

File: lstchain/reco/utils.py [hunk 9]
```diff
@@ -227,9 +236,13 @@ def camera_to_sky(pos_x, pos_y, focal, pointing_alt, pointing_az):
     focal = 28*u.m
     pointing_alt = np.array([1.0, 1.0]) * u.rad
     pointing_az = np.array([0.2, 0.5]) * u.rad
-    sky_coords = utils.camera_to_sky(pos_x, pos_y, focal, pointing_alt, pointing_az)
+    sky_coords = utils.camera_to_altaz_frame(pos_x, pos_y, focal, pointing_alt, pointing_az)
```
Comment (rlopezcoto) [inline lstchain/reco/utils.py:239]: `camera_to_altaz_frame` -> `camera_to_altaz`

**Comment by MaxNoe** [comment]
Is the obstime defined globally in the `utils.py` file used somewhere else? It should be removed.

**Comment by JouvinLea** [comment]
if you think it can be merge, I am personally done.
otherwise tell me what I should change.

Obstime at the beginning of the files is used to define the horizon_frame used in other methods . And I will not do this clean up in this PR. maybe one PR can be done to clean the utils.py in the future.

Once this is merge, I will do another PR for implementing ra/dec in the DL2 dataframe but other PR is better because this is another subject.
